airbyte-ci/connectors/live-tests/src/live_tests/commons/utils.py

```diff
@@ -84,22 +84,25 @@ async def get_container_from_dockerhub_image(dagger_client: dagger.Client, docke
         pytest.exit(f"Failed to import connector image from DockerHub: {e}")
 
 
-async def get_connector_container(dagger_client: dagger.Client, image_name_with_tag: str) -> dagger.Container:
+async def get_connector_container(dagger_client: dagger.Client, image_name_with_tag: str, is_ci: bool, is_target: bool) -> dagger.Container:
     """Get a dagger container for the connector image to test.
 
     Args:
         dagger_client (dagger.Client): The dagger client to use to import the connector image
         image_name_with_tag (str): The docker image name and tag of the connector image to test
+        is_ci (bool): Whether the tests are being run with airbyte-ci (locally or in GHA)
+        is_target (bool): Whether the container to get is the target container
 
     Returns:
         dagger.Container: The dagger container for the connector image to test
     """
     # If a container_id.txt file is available, we'll use it to load the connector container
     # We use a txt file as container ids can be too long to be passed as env vars
     # It's used for dagger-in-dagger use case with airbyte-ci, when the connector container is built via an upstream dagger operation
-    container_id_path = Path("/tmp/container_id.txt")
-    if container_id_path.exists():
-        return await get_container_from_id(dagger_client, container_id_path.read_text())
+    if is_ci and is_target:
+        container_id_path = Path("/tmp/container_id.txt")
```

Comment:
In order to not tie this utils to the regression test suite I'd suggest to make the airbyte-ci `RegressionTests` step write `/tmp/<docker-image-name-with-tag>_container_id.txt` and make this function check if such a file is available.

Reply:
I have a vague memory of having done that in an earlier version of the code 😄. Will wire it through. But, it is a slightly bigger lift so would you be open to approving this PR so we can unblock people and I'll do it in a follow up?

Reply:
Actually on second thought I think what you're proposing is slightly lower effort than what I was envisioning. Will do it now.

Reply:
Let me know if it's too much, happy to 👍  to unblock.

Reply:
Thanks @alafanechere I made this change.